src/xenia/kernel/xboxkrnl/xboxkrnl_rtl.cc

```diff
@@ -418,7 +418,10 @@ DECLARE_XBOXKRNL_EXPORT(RtlTryEnterCriticalSection,
                         ExportTag::kImplemented | ExportTag::kHighFrequency);
 
 void RtlLeaveCriticalSection(pointer_t<X_RTL_CRITICAL_SECTION> cs) {
-  assert_true(cs->owning_thread == XThread::GetCurrentThread()->guest_object());
+  if (cs->owning_thread != XThread::GetCurrentThread()->guest_object()) {
```

Comment:
You're right - games that do this are totally abusing the API but xboxkrnl will release the critical section regardless of the owner, so this if statement should be axed.

Reply:
I went and checked in the debugger and the actual issue is that the game is releasing a critical section that's not owned by _anyone_ (owning_thread=0, lock_count=0). I'll change this patch to a check for that, since it would just fail the assertion on line 424.

Reply:
Huh, that's interesting. Doing so natively will cause the recursion count to underflow and wrap around, which would corrupt the critical section.
Is the game misbehaving?

Reply:
Doesn't seem like it, the game plays fine. I set some conditional breakpoints and afaict nobody ever calls RtlEnterCriticalSection on that CS, so even if it is corrupted then maybe nobody ever noticed.